components/hab/src/command/cli/setup.rs

```diff
@@ -379,12 +379,10 @@ fn set_binlink_path(binlink_path: &Path) -> Result<()> {
             r"System\CurrentControlSet\Control\Session Manager\Environment",
             KEY_ALL_ACCESS,
         )?;
+        let mut paths = vec![binlink_path.to_path_buf()];
         let path: String = env.get_value("path")?;
-        let new_paths = [binlink_path, Path::new(&path)];
-        env.set_value(
-            "path",
-            &env::join_paths(new_paths.iter())?.to_str().unwrap(),
-        )?;
+        paths.append(&mut env::split_paths(&path).collect());
+        env.set_value("path", &env::join_paths(paths)?.to_str().unwrap())?;
```

Comment:
It took me a bit to get my head around this. Why not start with the vector containing the paths from the environment and add the binlink path to the front? Something like:
```rust
        let mut paths: Vec<PathBuf> = env::split_paths(&path).collect();
        paths.insert(0, Path::new("/bin/link").to_path_buf());
```